src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -819,6 +819,19 @@ private void pushFightLoopOnStack(final boolean firstRun) {
 
       @Override
       public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+        if (Match.someMatch(m_attackingUnits, Matches.unitHasAttackValueOfAtLeast(1))) {
```

Comment:
Potentially. Imagine I create a 'wall' unit that reduces enemy unit attack by 1. Then I attack with a unit that has only 1 attack so my total attack would reduce to 0.